From 296ff4f15244d6cace8ccf8061cebdef13281d05 Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@nxp.com>
Date: Mon, 13 Mar 2017 11:53:48 +0000
Subject: [PATCH 109/767] staging: fsl-dpaa2/eth: Disable preemption when
 draining pool

Prevent calling thread to be relocated on different CPU in the middle of
DPIO buffer acquire commands.

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index 544b08fe7601..34b580a87281 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -971,8 +971,10 @@ static void drain_pool(struct dpaa2_eth_priv *priv)
 {
 	int i;
 
+	preempt_disable();
 	drain_bufs(priv, DPAA2_ETH_BUFS_PER_CMD);
 	drain_bufs(priv, 1);
+	preempt_enable();
 
 	for (i = 0; i < priv->num_channels; i++)
 		priv->channel[i]->buf_count = 0;
-- 
2.17.0

